cpp/src/arrow/compute/exec/sink_node.cc

```diff
@@ -275,6 +302,11 @@ struct OrderBySinkNode final : public SinkNode {
   }
 
   void InputReceived(ExecNode* input, ExecBatch batch) override {
+    EVENT(span_, "InputReceived", {{"batch.length", batch.length}});
+    util::tracing::Span span;
+    START_SPAN_WITH_PARENT(span, span_, "InputReceived",
+                           {{"node.label", label()}, {"batch.length", batch.length}});
+
```

Comment:
From one of the screenshots above, it might be worth instrumenting DoFinish below for OrderBySinkNode since that's where the 'real' work happens.